### PR TITLE
XML Display for Collections

### LINKED ISSUE #107: Ensure that content negotiation exposes XML encoded finding aids
This is currently implemented in PULFA 2.0 in order to support the harvesting of collections and components by external organizations (please see https://findingaids.princeton.edu/collections/MC019/c01194 and https://findingaids.princeton.edu/collections/MC019/c01194.xml). A pattern is already structured for this in Blacklight with https://github.com/projectblacklight/blacklight/blob/master/lib/blacklight/routes/searchable.rb#L17, so this may be a minor enhancement.

We'll have to figure out how to save and manage these in a way that allows access to the EADs from multiple webservers.

## Implementation Proposal

* Create an AR Object with a text field, the call number string field, and an ID
* After downloading the XML stash it in postgres.
* For the pages just pull it from postgres if the record exists in SOLR.
* For component pages don't cache them, just parse it with Nokogiri and grab that component out.

### PR DESCRIPTION
**Pull request by tpendragon**
Closes #107

This adds an XML caching layer in ActiveRecord. It's the simplest low level cache store we have around and by utilizing one row per ead_id we don't have to worry about it getting too big or having a bunch of stale content.

### CODE REVIEW

File: app/models/solr_document.rb [hunk 1]
```diff
@@ -17,7 +17,8 @@ class SolrDocument
   # single valued. See Blacklight::Document::SemanticFields#field_semantics
   # and Blacklight::Document::SemanticFields#to_semantic_values
   # Recommendation: Use field names from Dublin Core
-  use_extension(Blacklight::Document::DublinCore)
+  # use_extension(Blacklight::Document::DublinCore)
+  SolrDocument.use_extension(Pulfalight::Document::XMLExport)
```
Comment (jrgriffiniii) [inline app/models/solr_document.rb:21]: 👍

**Comment by jrgriffiniii** [comment]
I'm assuming that there is no use case for cache expiry and refreshing the XML expression of the ArchivesSpace finding aid. Is this indeed the case?

**Comment by tpendragon** [comment]
> I'm assuming that there is no use case for cache expiry and refreshing the XML expression of the ArchivesSpace finding aid. Is this indeed the case?

I don't think so. The indexing job skips the cache, so every time it's indexed it should re-cache the material.

**Comment by faithc** [comment]
Is this only working for full the full finding aid and not individual pages/components?

**Comment by tpendragon** [comment]
I think there's a bug for some components, but the component display should be working.

**Comment by faithc** [comment]
Hmm, I've tried it in a couple of spots: https://findingaids.princeton.edu/catalog/C1491_c1.xml + https://findingaids.princeton.edu/catalog/C1491#description.xml.

**Comment by tpendragon** [comment]
@faithc Created https://github.com/pulibrary/pulfalight/issues/799. Please prioritize appropriately.

**Pull request merged**